simple_use_case/cwl/convert_msh_to_xdmf.cwl

```diff
@@ -2,10 +2,19 @@
 
 cwlVersion: v1.0
 class: CommandLineTool
-doc: |
-  Convert the produced gmsh mesh to xdmf
-baseCommand: ["meshio", "convert"]
-arguments: ["$(inputs.inputmesh.path)", "mesh_converted.xdmf"]
+doc: Convert the produced gmsh mesh to xdmf
+
+hints:
+  SoftwareRequirement:
+   packages:
+     meshio:
+       version: [ 5.0.5=pyhd8ed1ab_0, 5.0.5 ]
+       specs:
+         - https://anaconda.org/conda-forge/meshio
```

Comment:
Could you add some information which of the versions is chosen? Is the specs directive directly installing the conda package?

Reply:
Version & source is up to the CWL runner. For `cwltool`, the ` --beta-conda-dependencies` flag is described at https://github.com/common-workflow-language/cwltool#leveraging-softwarerequirements-beta (and in conda mode, `--beta-dependencies-directory` can be used to choose where to cache the created conda environments)